src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2027,6 +2030,8 @@ internal bool CheckValueKind(BoundExpression expr, BindValueKind valueKind, Diag
                 case BoundKind.DynamicMemberAccess:
                 case BoundKind.DynamicIndexerAccess:
                     return CheckDynamicValueKind(expr, valueKind, diagnostics);
+                case BoundKind.NotNullExpression:
+                    return CheckValueKind(((BoundNotNullExpression)expr).Expression, valueKind, diagnostics);
```

Comment:
>return CheckValueKind(((BoundNotNullExpression)expr).Expression, valueKind, diagnostics); [](start = 20, length = 89)

I believe ```private BoundExpression CheckValue(BoundExpression expr, BindValueKind valueKind, DiagnosticBag diagnostics)``` should also be changed to dig through the ```BoundNotNullExpression``` and preserve any transformations made to the underlying expression in the process.